### ISSUE TITLE
ImportError: cannot import name 'linux_distribution' from 'platform' (/usr/lib/python3.8/platform.py)

### ISSUE THREAD

**Issue by haithcockce** [RedHatInsights/insights-core#2964]
It looks as if there's some reliance on `platform.linux_distribution` which was removed in [Python 3.8](https://docs.python.org/2/library/platform.html#platform.linux_distribution) so `pytest` fails when running Python 3.8 in the environment: 

```bash
 ~/projects/insights-core
 pytest
===================================================== test session starts =====================================================
platform linux -- Python 3.8.5, pytest-3.0.6, py-1.10.0, pluggy-0.4.0
rootdir: /home/chaithco/projects/insights-core, inifile: setup.cfg
plugins: cov-2.4.0
collected 1830 items / 7 errors 

=================================================== short test summary info ===================================================
ERROR insights/tests/client/test_net_comm.py
ERROR insights/tests/client/apps/test_compliance.py
ERROR insights/tests/client/phase/test_LEGACY_post_update.py
ERROR insights/tests/client/phase/test_collect_and_upload.py
ERROR insights/tests/client/phase/test_post_update.py
ERROR insights/tests/client/phase/test_pre_update_support.py
ERROR insights/tests/client/phase/test_update.py
=========================================================== ERRORS ============================================================
___________________________________ ERROR collecting insights/tests/client/test_net_comm.py ___________________________________
ImportError while importing test module '/home/chaithco/projects/insights-core/insights/tests/client/test_net_comm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
insights/tests/client/test_net_comm.py:3: in <module>
    from insights.client.phase import v1
insights/client/phase/v1.py:15: in <module>
    from insights.client.apps.compliance import ComplianceClient
insights/client/apps/compliance/__init__.py:6: in <module>
    from platform import linux_distribution
E   ImportError: cannot import name 'linux_distribution' from 'platform' (/usr/lib/python3.8/platform.py)
_______________________________ ERROR collecting insights/tests/client/apps/test_compliance.py ________________________________
ImportError while importing test module '/home/chaithco/projects/insights-core/insights/tests/client/apps/test_compliance.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
insights/tests/client/apps/test_compliance.py:3: in <module>
    from insights.client.apps.compliance import ComplianceClient, OSCAP_RESULTS_OUTPUT, COMPLIANCE_CONTENT_TYPE
insights/client/apps/compliance/__init__.py:6: in <module>
    from platform import linux_distribution
E   ImportError: cannot import name 'linux_distribution' from 'platform' (/usr/lib/python3.8/platform.py)
```

And so on. 

It looks like the recommendation is to migrate to the [`distro`](https://pypi.org/project/distro/) module to get the same functionality. While I have not tested downgrading to 3.7 to see if the issue is reproducible, the documentation is pretty clear `platform.linux_distribution` so I don't see a super strong reason to attempt reproducing on 3.7 or below.

**Comment by ryan-blakley** [comment]
@gravitypriest it looks like this issue was corrected in #3049, could you confirm if that is correct?

**Comment by gravitypriest** [comment]
@ryan-blakley yep! truth be told I forgot there was an open issue about it. We replaced it with some insights-core spec functions.
